Add terraform postgres resources to Duchy worker2

YuhongWang-Amazon opened the pull request:
This PR adds the postgres resources to duchy worker2. Both postgres db and spanner db will co-exists until it is validated that postgres duchy works well with the other spanner duchies.

YuhongWang-Amazon committed:
Change worker2 duchy terraform to use postgres

YuhongWang-Amazon committed:
Change worker2 duchy terraform to use postgres

wfa-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1134)
<!-- Reviewable:end -->

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
refactor

YuhongWang-Amazon committed:
refactor

YuhongWang-Amazon committed:
refactor

YuhongWang-Amazon committed:
refactor

YuhongWang-Amazon committed:
Merge branch 'main' into yuhong.postgres-duchy-terraform

YuhongWang-Amazon reviewed (commented):
terraform plan result: https://github.com/world-federation-of-advertisers/cross-media-measurement/actions/runs/5658308391/job/15329319645

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1134#-:-NaInEIUDewYwAtddyT8:bumdede)* status: 0 of 13 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

SanjayVas committed:
Allow SyntheticGenerator to be used for correctness test on GKE. (#1127)

SanjayVas committed:
Use SyntheticGenerator instead of BigQuery for K8s test workflow. (#1128)

bdomen-ggl committed:
Move UI src and test to experimental folder (#1139)

Moving the Reporting UI to an experimental folder

bdomen-ggl committed:
adding UI assets (#1141)

Adding assets for Reporting UI - font and icons.

Marco-Premier committed:
Update cross-media-measurement-api version (#1123)

Replace Timestamps with google.type.Date in ModelRolloutService Fix
broken tests

Internal proto resources remained the same. What changed is the way
public proto resources are converted into internal ones and viceversa.

---------

Co-authored-by: marcopremier <marcopremier@meta.com>

renjiezh committed:
Duchy Api and System Api support Reach-Only Llv2 protocol. (#1125)

SanjayVas committed:
Read synthetic generator specs from files. (#1142)

This allows the spec for a given EDP simulator to be configurable.

tristanvuong2021 committed:
Drop some metric labels for GCP (#1140)

iverson52000 committed:
Update PBM InMemoryBackingStore and PostgresBackingStore for ACDP com… (#1137)

…position

Rally task link:
https://rally1.rallydev.com/#/?detail=/userstory/702734912681&fdp=true

SanjayVas committed:
Add support for protobuf Duration and Timestamp fields in synthetic generator. (#1145)

renjiezh committed:
Herald supports reach-only llv2 protocol. (#1135)

YuhongWang-Amazon committed:
Add postgres computations service (#1136)

This adds the postgres computations service and unit tests for it.

SanjayVas committed:
Regenerate expired Reporting TLS cert. (#1149)

The new certificate is valid for 10 years.

ple13 committed:
Implementation of the initialization, setup, and execution phase of the reach-only protocol. (#1129)

Implement the phases of the new sparse reach only protocol:

1. Initialization phase: Duchies sample local El Gamal keypair.
2. Setup phase:
a) Non-aggregators add noise registers to the crv and shuffle the
modified crv. They encrypt their excessive noise using the composite El
Gamal public key. They send the crv and the excessive noise ciphertext
to the aggregator.
b) Aggregator: Waits for the crv and the excessive noise ciphertexts
from non-aggregators. It adds noise to the crv, shuffles it, and
encrypts its excessive noise with the composite El Gamal key. It then
combines the excessive noise ciphertexts by adding them together. It
sends the modified crv and the excessive noise ciphertext to the next
worker.
3. Execution phase: Duchies collaborate to decrypt, randomize the
register indices, and decrypt the excessive noise ciphertext. The
aggregator can count the number of distinct registers, obtain the total
amount of excessive noise, then estimate the reach based on the
available information.

iverson52000 committed:
Update PrivacyBudgetLedger for ACDP composition (#1148)

SanjayVas committed:
Add a larger-scale SyntheticEventGroupSpec. (#1144)

This is used in the dev K8s configuration. It is not used in local tests due to CI memory constraints.

SanjayVas committed:
Remove non-template fields from our test templates. (#1150)

Every field on an event message should use a message type that has an event_template annotation, and every field on a template message should have a template_field annotation.

iverson52000 committed:
Update PrivacyBudgetManager for ACDP composition (#1157)

YuhongWang-Amazon committed:
Add in process postgres duchy correctness tests (#1152)

This adds in process postgres duchy correctness tests

renjiezh committed:
EdpSimulator fulfills Reach-Only LiquidLegionsV2 requisitions. (#1146)

YuhongWang-Amazon committed:
Add empty cluster test for postgres duchy

YuhongWang-Amazon committed:
fix

YuhongWang-Amazon committed:
factor

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
update

YuhongWang-Amazon committed:
fix bug

YuhongWang-Amazon committed:
add missing images and runners

YuhongWang-Amazon committed:
format

YuhongWang-Amazon committed:
bring back unit tests

YuhongWang-Amazon committed:
fix bug

YuhongWang-Amazon committed:
update dev duchy worker2

YuhongWang-Amazon committed:
Merge branch 'yuhong.postgres-duchy-terraform' of github.com:world-federation-of-advertisers/cross-media-measurement into yuhong.postgres-duchy-terraform
